examples/StreamLoop.cpp

```diff
@@ -0,0 +1,60 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// Copyright 2020 Lawrence Livermore National Security, LLC and other CARE developers.
+// See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//////////////////////////////////////////////////////////////////////////////////////
+
+// This macro needs to be defined before including care/care.h,
+// which allows you to port a file at a time. Without this define,
+// LOOP_STREAM will run on the CPU. With this define, LOOP_STREAM
+// will run on the GPU only if you built with CUDA enabled. Otherwise,
+// if CUDA is disabled, it will default back to running on the host.
+#define GPU_ACTIVE
+
+// CARE library header
+#include "care/care.h"
+#include "care/array_utils.h"
+
+// Std library headers
+#include <algorithm>
+#include <iostream>
+#include <limits>
+#include <vector>
+
+int main(int, char**) {
+   // Normal code
+   int size = 1000;
+   std::vector<int> data(size);
+
+   for (int i = 0; i < size; ++i) {
+      data[i] = i * i;
+   }
+
+   int max = *std::max_element(data.begin(), data.end());
+   std::cout << "CPU max: " << max << std::endl;
+
+   // Ported code
+   int size2 = 1000;
+   care::host_device_ptr<int> data2(size2);
+
+   LOOP_STREAM(i, 0, size2) {
+      data2[i] = i * i;
```

Comment:
Likewise in this file and elsewhere, be very generous with comments in examples

Reply:
Done.